### DISCUSSION TITLE
What do you think of discussions?

### DISCUSSION THREAD

**Discussion by chrisjsewell** [aiidateam/aiida-core#4444] [Q&A]
Hey all!

GitHub Discussions is a new feature which is in [Beta development](https://github.blog/2020-05-06-new-from-satellite-2020-github-codespaces-github-discussions-securing-code-in-private-repositories-and-more/#discussions), of which I am part of the beta testers (we have been using it for a while on https://github.com/executablebooks/meta/discussions).

Personally, I feel it is a lot nicer alternative to our current https://groups.google.com/g/aiidausers forum, offering nicer integration with the repository; it's easier to reference code, issues etc, and you can even transfer discussions to issues and visa-versa.

A key current limitation I see, is that you can not make announcements as easily, to send out to users (like version releases), or allow users to only watch the repo for new discussion posts.
This is already something I have raised with them, and I believe they are working on a solution (https://github.com/gh-community/discussions-feedback/issues/113#issuecomment-668710219).
Of note though, you can spotlight certain discussions (up to 3), as I have done with this one.

So yes, let me know what you think and I so can pass on feedback to them! 😄 

Cheers, Chris

**Comment by chrisjsewell** [comment]
cc @giovannipizzi @sphuber @CasperWA @ltalirz @csadorf @mbercx @yakutovicha @ramirezfranciscof

**Comment by ltalirz** [comment]
I've been waiting for it to be released to give it a try.
Is there a way to import the content of a google group?

**Reply by chrisjsewell** [reply]
Well wait no more lol!

> Is there a way to import the content of a google group?

Not yet 😞, but there is an open discussion on "Migration from Google Groups"; to quote one of the github team:

> We don't have an API available yet but it's going to be one of our top priorities right after we 🚢 the limited public beta as a public beta.

**Comment by greschd** [comment]
What's your feeling for when something should be a discussion here, vs. on Slack?

Concerning this vs. the Google group - is there any integration with E-Mail? The Google group in my mind is just a mailing list; I think for many "casual" users that don't visit Github on the daily that's still the more accessible medium.

**Reply by chrisjsewell** [reply]
vs Slack, my feeling is that everything that can be made public should be, especially any discussion that will be helpful for the wider community. 
For example we are now going to make our bi-weekly developer meeting notes public

**Reply by chrisjsewell** [reply]
as shown below (https://github.com/aiidateam/aiida-core/discussions/4444#discussioncomment-99561) you can already make replies by email.
As I note though, users cannot yet only follow discussions (and not be emailed e.g. about issues etc), but hopefully that will be fixed soon(ish).
I guess this approach also necessitates that users have a GitHub, although I imagine most users of AiiDA will.

For sure, there is pros and cons to discuss though

**Reply by chrisjsewell** [reply]
Just the fact that you can write syntax highlighted python, which you can't do in google groups or ~~slack~~ (ah I just found out they have finally added this, although its still a bit of a hassle), is a big win IMO lol

```python
# This program prints Hello, world!

print('Hello, world!')
```

**Reply by greschd** [reply]
Right - my experience with other "web first" services is just that these features can end up with a less-than-pleasant experience for the users that prefer e-mail.

Formatting can probably work, but e.g. editing a post doesn't really translate. All that's to say, we should probably keep the mailing list around.

**Reply by chrisjsewell** [reply]
How do you edit properly with the mailing list? Every time I email the mailing list the formatting turns out awful lol.
Obviously different people have different preferences, but personally I find the whole mailing list experience awful, it just clogs up my email with rubbish.
Anyway aren't you meant to be team Microsoft 😆

**Reply by greschd** [reply]
Plain text, baby.

Point here is, e-mail is already in everybody's tool-set, and you can configure it to your liking (filters, sorting, what have you). Personally, I'd rather live with some less-fancy formatting than having to check 10 different chat apps.

[All opinions are my own, not those of my employer 😎]

**Reply by chrisjsewell** [reply]
Well, to get the last word in, jupyter-book and jupyter (which uses discourse) have much more users than aiida, and they manage fine without a crappy google mailing list 😝

**Reply by greschd** [reply]
https://www.python.org/community/lists/  ¯\\_(ツ)_/¯

**Comment by chrisjsewell** [comment]
This is a reply via email

**Comment by csadorf** [comment]
Interesting feature. Certainly very interested in trying it out. However, I think we should be careful with replacing the groups, because they have a pretty large reach at the moment afaik.

**Reply by giovannipizzi** [reply]
I agree, I would make a very careful balance before replacing the mailing list - the use case is different, and the email should remain and be limited to important messages - even if you can probably subscribe to discussions, I already envision them (see also this thread already ;-) ) to become closer to a chat than a mailing list, with the risk of "spamming" non-developers.

**Reply by chrisjsewell** [reply]
Certainly we should consider the use cases 👍 
I would note that you can easily Unsubscribe from individual discussions, which is more difficult with the mailing list

**Reply by espenfl** [reply]
That is indeed a very useful feature.

**Comment by chrisjsewell** [comment]
FYI, it is now possible for people to only get notifications for discussions (as opposed to issues & PRs):

<img width="332" alt="image" src="https://user-images.githubusercontent.com/2997570/100460293-3a51da00-30c7-11eb-8bf7-ed99c44f6035.png">
